script.js

```diff
@@ -1,4 +1,175 @@
+
+var playerNo = 1;
+var player1Dice = [];
+var player2Dice = [];
+var player1CombinedNo;
+var player2CombinedNo;
+var rolledDice = false;
+var player1Score = 0;
+var player2Score = 0;
+
+var isGameModeSet = false;
+var gameMode = "normal";
+
+var playerRollDice = function (playerNum, playerDice) {
+  var diceRoll1 = diceRoll();
+  var diceRoll2 = diceRoll();
+  playerDice[0] = diceRoll1;
+  playerDice[1] = diceRoll2;
+
+  rolledDice = true;
+  return (
+    "player " +
+    playerNum +
+    " rolled " +
+    playerDice[0] +
+    " and " +
+    playerDice[1] +
+    ", please pick the dice order. "
+  );
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (!isGameModeSet) {
+    if (input === "reverse") {
+      gameMode = "reverse";
+      return "The game is in Reverse Mode";
+    }
+    isGameModeSet = true;
+  }
+
+  if (playerNo == 1 && !rolledDice) {
+    return playerRollDice(playerNo, player1Dice);
+  }
+
+  if (playerNo == 1 && rolledDice) {
+    //validation
+    if (!(input === "1" || input === "2")) {
+      return "please input 1 or 2 to decide the order of the dice.";
+    }
+
+    var playerInputNumber = getPlayerNumber(input, player1Dice);
+    player1CombinedNo = playerInputNumber;
+
+    player1Score += playerInputNumber;
+
+    playerNo = 2;
+    rolledDice = false;
+
+    return (
+      "Player 1 combined number is " +
+      player1CombinedNo +
+      ". It is Player 2 turn now."
+    );
+  }
+
+  if (playerNo == 2 && !rolledDice) {
+    return playerRollDice(playerNo, player2Dice);
+  }
+  //validation
+  if (playerNo == 2 && rolledDice) {
+    if (!(input === "1" || input === "2")) {
+      return "please input 1 or 2 to decide the order of the dice.";
+    }
+    var playerInputNumber = getPlayerNumber(input, player2Dice);
+    player2CombinedNo = playerInputNumber;
+    var winningOutput = winningCondition();
+    var reverseWinningOutput = reverseWinningCondition();
+
+    player2Score += playerInputNumber;
+    var playerLeader = leaderboard();
+
+    //reset
+    rolledDice = false;
+    playerNo = 1;
+
+    if (gameMode == "reverse") {
+      isGameModeSet = true;
+      outputValue =
+        "Player 1 combined number is " +
+        player1CombinedNo +
+        ". Player 2 combined number is " +
+        player2CombinedNo +
+        "." +
+        reverseWinningOutput;
+      return outputValue;
+    } else {
+      if (gameMode == "normal") {
+        isGameModeSet = false;
+
+        outputValue =
+          "Player 1 combined number is " +
+          player1CombinedNo +
+          ". Player 2 combined number is " +
+          player2CombinedNo +
+          "." +
+          winningOutput +
+          "<br>" +
+          playerLeader;
+        return outputValue;
+      }
+    }
+  }
+  //leaderboard condition
+  function leaderboard() {
```

Comment:
It's good practice to group all your helper functions together and put the main function at the bottom.